compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -311,15 +308,18 @@ static void build_accessor(AggregateType* ct, Symbol* field, bool setter) {
 
   fn->addFlag(FLAG_FIELD_ACCESSOR);
 
-  if (fieldIsConst)
+  if (fieldIsConst && !typeMethod)
     fn->addFlag(FLAG_REF_TO_CONST);
-  else if (recordLike)
+  else if (recordLike && !typeMethod)
```

Comment:
I would have the typeMethod check be unified outside rather than added to both, i.e.:

```
if (!typeMethod) {
  if (fieldIsConst)
    fn->addFlag(FLAG_REF_TO_CONST);
  else if (recordLike)
  // etc.
```